docs/reference/run.md

```diff
@@ -1714,13 +1714,21 @@ $ docker run -d --tmpfs /run:rw,noexec,nosuid,size=65536k my_image
 ### VOLUME (shared filesystems)
 
     -v, --volume=[host-src:]container-dest[:<options>]: Bind mount a volume.
-    The comma-delimited `options` are [rw|ro], [z|Z],
+    The comma-delimited `options` are [rw|ro|ro-non-recursive|ro-force-recursive|rro], [z|Z],
```

Comment:
I'm on the fence if we want the advanced options on this (we currently also don't have `bind-nonrecursive`).


IIUC, currently we consider `ro` to be "recursive read-only" (best effort) on v25.0.0, and "read-only" (non-recursive) on v24.0.0 and lower.

- How would `rro` and `ro-force-recursive` differ?
- `ro-non-recursive` would be for keeping the "pre-v25.0.0" behavior (read-only, but not for submounts?)

For the last one, I think that would _only_ apply to implicit submounts (i.e., I bind-mount `/foo` and there happens to be a `/foo/some-mount` present, which is a mount), but it would _not_ apply to an explicit mount inside such a path, so wouldn't apply to;

```bash
docker run \ 
  - v /foo:/foo:ro
  -v /bla:/foo/some-mount:rw \
 busybox
```

Reply:
These are defined on the daemon side:
https://github.com/moby/moby/blob/78cf11575dcf83092cbad255c49ab1e8b7f00d4d/volume/mounts/parser.go#L14-L21

> How would rro and ro-force-recursive differ?

Same.

> ro-non-recursive would be for keeping the "pre-v25.0.0" behavior (read-only, but not for submounts?)

Yes.

Reply:
(But maybe want rename `ro-non-recursive` to `ro-nonrecursive`, for consistency with `bind-nonrecursive` in the CSV form)

Reply:
> These are defined on the daemon side:
> https://github.com/moby/moby/blob/78cf11575dcf83092cbad255c49ab1e8b7f00d4d/volume/mounts/parser.go#L14-L21

Arf.. that legacy API keeps haunting me. I'd like to move away from moving that API altogether (and do conversion on the client-side to use the more structured new API.

I think these options are "advanced" options, and for those, I'd rather use the `--mount` CSV flags, and keep the short-hand flags for the common cases. The shorthand micro-format is already overloaded.